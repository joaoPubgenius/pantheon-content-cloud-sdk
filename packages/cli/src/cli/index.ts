#!/usr/bin/env node
import { exit } from 'process';
import yargs from 'yargs';
import { hideBin } from 'yargs/helpers';
import init from './commands/init';
import { createToken, listTokens, revokeToken } from './commands/token';
import login from './commands/login';
import logout from './commands/logout';
import { createSite, listSites, updateSite } from './commands/sites';

yargs(hideBin(process.argv))
  .scriptName('pcc')
  .usage('$0 <cmd>')
  .strictCommands()
  .demandCommand()
  .command(
    'init <project_directory> [options]',
    'Sets up project with required files.',
    (yargs) => {
      yargs
        .positional('<project_directory>', {
          describe: 'The project directory in which setup should be done.',
          demandOption: true,
          type: 'string',
        })
        .option('template', {
          describe: 'Template from which files should be copied.',
          type: 'string',
          choices: ['nextjs', 'gatsby'],
          demandOption: true,
        });
    },
    async (args) => {
      const dirName = args.project_directory as string;
      const template = args.template as CliTemplateOptions;

      await init({ dirName, template });
    },
  )
  .command(
    'token <cmd> [options]',
    'Enables you to manage tokens for a PCC project.',
    (yargs) => {
      yargs
        .strictCommands()
        .demandCommand()
        .command(
          'create',
          'Creates new token.',
          (yargs) => {},
          async (args) => await createToken(),
        )
        .command(
          'list',
          'Lists existing tokens.',
          (yargs) => {},
          async (args) => await listTokens(),
        )
        .command(
          'revoke <id>',
          'Revokes token for a given id.',
          (yargs) => {
            yargs.positional('<id>', {
              describe: 'ID of the token which you want to revoke',
              demandOption: true,
              type: 'string',
            });
          },
          async (args) => await revokeToken(args.id as string),
        );
    },
    async (args) => {},
  )
  .command(
    'site <cmd> [options]',
    'Enables you to manage sites for a PCC project.',
    (yargs) => {
      yargs
        .strictCommands()
        .demandCommand()
        .command(
          'create [options]',
          'Creates new site.',
          (yargs) => {
            yargs.option('url', {
              describe: 'Site url',
              type: 'string',
              demandOption: true,
            });
          },
          async (args) => await createSite(args.url as string),
        )
        .command(
          'list',
          'Lists existing sites.',
          (yargs) => {},
          async (args) => await listSites(),
        )
        .command(
          'update <id> [options]',
          'Updates site for a given ID.',
          (yargs) => {
            yargs
              .positional('<id>', {
                describe: 'ID of the site which you want to update',
                demandOption: true,
                type: 'string',
              })
              .option('url', {
                describe: 'Site url',
                type: 'string',
              });
          },
          async (args) =>
            await updateSite({
              id: args.id as string,
              url: args.url as string,
            }),
        );
    },
    async (args) => {},
  )
  .command(
    'login',
    'Logs you in you to PCC client.',
    (yargs) => {},
    async (args) => await login(),
  )
  .command(
    'logout',
    'Logs you out you from PCC client.',
    (yargs) => {},
    async (args) => await logout(),
  )
  .help(true).argv;
