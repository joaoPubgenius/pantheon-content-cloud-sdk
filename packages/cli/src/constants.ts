import os from 'os';
import path from 'path';

export const PCC_ROOT_DIR = path.join(os.homedir(), '.pcc');
