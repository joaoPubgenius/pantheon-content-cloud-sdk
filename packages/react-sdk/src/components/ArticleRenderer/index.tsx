import React from 'react';
import { ReactMarkdown } from 'react-markdown/lib/react-markdown.js';
import rehypeRaw from 'rehype-raw';

import { Article } from '../../types';

import ArticleComponent from './ArticleComponent';
import TopLevelElement from './TopLevelElement';

interface Props {
  article?: Article;
  bodyClassName?: string;
  containerClassName?: string;
  headerClassName?: string;
  renderTitle?: (titleElement: React.ReactElement) => React.ReactNode;
}

const ArticleRenderer = ({
  article,
  headerClassName,
  bodyClassName,
  containerClassName,
  renderTitle,
}: Props) => {
  const contentType = article?.contentType;

  if (contentType === 'TEXT_MARKDOWN') {
    return (
      <div className={containerClassName}>
        {article?.content ? (
          <ReactMarkdown rehypePlugins={[rehypeRaw]}>
            {String(article.content)}
          </ReactMarkdown>
        ) : (
          <span>No content to display</span>
        )}
      </div>
    );
  }

  const parsedBody: any[] = article?.content ? JSON.parse(article.content) : [];
  const indexOfFirstParagraph = parsedBody.findIndex((x) =>
    ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'h7', 'p'].includes(x.tag),
  );

  const [titleElement] = parsedBody.splice(indexOfFirstParagraph, 1);

  const titleComponent = titleElement ? (
    <ArticleComponent x={titleElement.children} />
  ) : (
    <span>{article?.title}</span>
  );

  return (
    <div className={containerClassName}>
      <div className={headerClassName}>
        {renderTitle ? renderTitle(titleComponent) : titleComponent}
      </div>
      <div className={bodyClassName}>
        {parsedBody?.map((x: any, idx) => (
          // No stable key available
          // eslint-disable-next-line react/no-array-index-key
          <TopLevelElement element={x} key={idx} keyElem={idx} />
        ))}
      </div>
    </div>
  );
};

export default ArticleRenderer;
