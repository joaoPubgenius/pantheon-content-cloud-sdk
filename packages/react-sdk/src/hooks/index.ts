export { useArticle } from './use-article';
export { useArticles } from './use-articles';
