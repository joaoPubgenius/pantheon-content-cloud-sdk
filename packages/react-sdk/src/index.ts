export { PantheonProvider } from "./core/pantheon-context";
export { PantheonClient } from "@pantheon-systems/pcc-sdk-core";

export {
  getArticles,
  getArticle,
  getAllTags,
} from "@pantheon-systems/pcc-sdk-core";

// Would be nice to have this as a separate entry point but
// context is defined at module scope so can't be shared between
// entry points.
// https://github.com/facebook/react/issues/19541
export * from "./hooks";

export * from "./types";
