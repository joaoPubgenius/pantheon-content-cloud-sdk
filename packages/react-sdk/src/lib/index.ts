export { getArticles, getArticle } from './articles';
export { getAllTags } from './metadata';
