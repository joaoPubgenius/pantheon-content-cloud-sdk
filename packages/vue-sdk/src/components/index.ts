export { default as ArticleRenderer } from "./Renderer";
