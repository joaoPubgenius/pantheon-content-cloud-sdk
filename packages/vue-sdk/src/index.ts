export { plugin as pccPlugin } from "./plugin";

export type { PantheonClientConfig } from "./core/pantheon-client";

export * from "./hooks";

export * from "./types";

export * from "./components";
