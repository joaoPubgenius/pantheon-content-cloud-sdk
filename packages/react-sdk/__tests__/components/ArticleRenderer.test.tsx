import renderer from 'react-test-renderer';

import { Article } from '../../src';
import { ArticleRenderer } from '../../src/components';
import article from '../data/article.json';

describe('<ArticleRenderer />', () => {
  it("should render a post's content", () => {
    const tree = renderer
      .create(<ArticleRenderer article={article as Article} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
