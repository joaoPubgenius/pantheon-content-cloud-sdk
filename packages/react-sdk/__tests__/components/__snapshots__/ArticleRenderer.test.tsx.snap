// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleRenderer /> should render a post's content 1`] = `
<div>
  <div>
    <span
      style={{}}
    >
      This is a title, there are many like it, but this one is mine
    </span>
  </div>
  <div>
    <h1>
      <span
        style={{}}
      >
        This is h1
      </span>
    </h1>
    <h2>
      <span
        style={{}}
      >
        This is h2
      </span>
    </h2>
    <h3>
      <span
        style={{}}
      >
        This is h3
      </span>
    </h3>
    <h4>
      <span
        style={{}}
      >
        This is h4
      </span>
    </h4>
    <h5>
      <span
        style={{}}
      >
        This is h5
      </span>
    </h5>
    <h6>
      <span
        style={
          {
            "fontStyle": "italic",
          }
        }
      >
        This is h6
      </span>
    </h6>
    <p>
      <span
        style={{}}
      />
    </p>
    <p>
      <span
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        This is bold
      </span>
    </p>
    <p>
      <span
        style={
          {
            "fontStyle": "italic",
          }
        }
      >
        This is italic
      </span>
    </p>
    <p>
      <span
        style={
          {
            "textDecoration": "underline",
          }
        }
      >
        This is underlined
      </span>
    </p>
    <p>
      <span
        style={
          {
            "textDecoration": "underline",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "super",
          }
        }
      >
        aaa
      </span>
      <span
        style={
          {
            "fontStyle": "italic",
            "fontWeight": "bold",
            "textDecoration": "underline",
            "verticalAlign": "super",
          }
        }
      >
        This is bold, underlined, italic, strikethrough and supertext
      </span>
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        eee
      </span>
      <span
        style={
          {
            "fontStyle": "italic",
            "fontWeight": "bold",
            "textDecoration": "underline",
            "verticalAlign": "sub",
          }
        }
      >
        This is bold, underlined, italic, strikethrough and subscript
      </span>
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        horizontal line below:
      </span>
    </p>
    <hr />
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={{}}
      />
    </p>
    <p>
      <span
        style={{}}
      >
        x
      </span>
      <span
        style={
          {
            "verticalAlign": "super",
          }
        }
      >
        3
      </span>
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "super",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "super",
          }
        }
      >
        H
      </span>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        2
      </span>
      <span
        style={
          {
            "verticalAlign": "super",
          }
        }
      >
        O x
      </span>
      <span
        style={
          {
            "fontWeight": "bold",
            "verticalAlign": "super",
          }
        }
      >
        o
      </span>
    </p>
    <p>
      <span
        style={{}}
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        Unordered “bullet point” list
      </span>
    </p>
    <ul>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Apple
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Pineapple
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Pen 
        </span>
        <span
          style={
            {
              "fontWeight": "bold",
              "verticalAlign": "sub",
            }
          }
        >
          pineapple
        </span>
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
           
        </span>
        <span
          style={
            {
              "fontStyle": "italic",
              "verticalAlign": "sub",
            }
          }
        >
          apple
        </span>
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
           
        </span>
        <span
          style={
            {
              "textDecoration": "underline",
              "verticalAlign": "sub",
            }
          }
        >
          pen
        </span>
      </li>
    </ul>
    <p>
      <span
        style={
          {
            "textDecoration": "underline",
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        Unordered list with different symbols
      </span>
    </p>
    <ul>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Duck
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Duck2
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Goose
        </span>
      </li>
    </ul>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        Todo list with checkmarks
      </span>
    </p>
    <ul>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Veni
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Vidi
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Vici
        </span>
      </li>
    </ul>
    <p>
      <span
        style={
          {
            "textDecoration": "underline",
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        Ordered “numbered” list
      </span>
    </p>
    <ol>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Waking in the morning
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Gotta go downstairs
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Gotta have cereal
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          I see my friends
        </span>
      </li>
      <li
        style={{}}
      >
        <span
          style={
            {
              "verticalAlign": "sub",
            }
          }
        >
          Gotta get down on Friday
        </span>
      </li>
    </ol>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={{}}
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "fontWeight": "bold",
            "verticalAlign": "sub",
          }
        }
      >
        A wild table appears
      </span>
    </p>
    <table
      className="text-black"
    >
      <tbody>
        <tr
          style={{}}
        >
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                Merged cols 1 & 2
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                Col 3
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                Col 4
              </span>
            </p>
          </td>
        </tr>
        <tr
          style={{}}
        >
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                A
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                B
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                C
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                D
              </span>
            </p>
          </td>
        </tr>
        <tr
          style={{}}
        >
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                ა
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                ბ
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                ს
              </span>
            </p>
          </td>
          <td
            style={{}}
          >
            <p>
              <span
                style={
                  {
                    "verticalAlign": "sub",
                  }
                }
              >
                დ
              </span>
            </p>
          </td>
        </tr>
      </tbody>
    </table>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      />
    </p>
    <p>
      <span
        style={
          {
            "verticalAlign": "sub",
          }
        }
      >
        Some more paragraphs after everything else.
      </span>
    </p>
    <p>
      <span
        style={{}}
      />
    </p>
    <p>
      <span
        style={{}}
      />
    </p>
  </div>
</div>
`;
